### PR TITLE
Remove deprecated constraints and dynamic dim support with test fixes

### PR DESCRIPTION
**Pull request by IanNod**
The backing APIs have been deprecated for a year and were removed from PyTorch 2.6 nightlies, causing import errors.

torch.export and torch-mlir have long since provided a supported way for exporting dynamic dims.

### CODE REVIEW

**Comment by saienduri** [comment]
With this, we can remove the torch restriction here: https://github.com/iree-org/iree-turbine/blob/main/pytorch-rocm-requirements.txt#L3

**Review by saienduri — APPROVED** [review]
